In [1]:
from autogen_agentchat.agents import AssistantAgent
from autogen_ext.models.ollama import OllamaChatCompletionClient

In [2]:
ollama_model_client = OllamaChatCompletionClient(model="llama3.1")

## **Agent Define**
### ***Agent need mainly two things***
- Agent Name
- Agent Brain

In [3]:
agent_1 = AssistantAgent(
    name="My_Assistant",
    model_client=ollama_model_client,
    system_message="You are a helpful ai assistant. you are a math solver."
)

In [4]:
response = await agent_1.run(
    task = "Hi, my i'm Al Amin, How are you!!"
)

In [6]:
response.messages[-1]

TextMessage(id='61af2d3d-8c40-47b6-be00-50bb05a6f08c', source='My_Assistant', models_usage=RequestUsage(prompt_tokens=41, completion_tokens=107), metadata={}, created_at=datetime.datetime(2025, 7, 15, 5, 25, 9, 161693, tzinfo=datetime.timezone.utc), content="As-salamu alaykum, Al Amin! I'm doing great, thank you for asking! It's always wonderful to meet new people and assist with their math-related queries.\n\nI'm here to help you with any math problems or questions you may have. Whether it's algebra, geometry, calculus, or anything else, feel free to share the problem and I'll do my best to guide you through it.\n\nWhat brings you here today? Do you have a specific math problem you'd like some assistance with?", type='TextMessage')

TextMessage(id='61af2d3d-8c40-47b6-be00-50bb05a6f08c', source='My_Assistant', models_usage=RequestUsage(prompt_tokens=41, completion_tokens=107), metadata={}, created_at=datetime.datetime(2025, 7, 15, 5, 25, 9, 161693, tzinfo=datetime.timezone.utc),
 
content="As-salamu alaykum, Al Amin! I'm doing great, thank you for asking! It's always wonderful to meet new people and assist with their math-related queries.\n\nI'm here to help you with any math problems or questions you may have. Whether it's algebra, geometry, calculus, or anything else, feel free to share the problem and I'll do my best to guide you through it.\n\nWhat brings you here today? Do you have a specific math problem you'd like some assistance with?", 

type='TextMessage')

In [7]:
await agent_1.run(
    task= "What was my last question?"
)

TaskResult(messages=[TextMessage(id='a9419b76-770d-45a2-88ce-b8f8c58cee23', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 15, 5, 29, 5, 497285, tzinfo=datetime.timezone.utc), content='What was my last question?', type='TextMessage'), TextMessage(id='3da3b1a9-ed9f-4a0b-b385-fb2d6b7ccf60', source='My_Assistant', models_usage=RequestUsage(prompt_tokens=163, completion_tokens=44), metadata={}, created_at=datetime.datetime(2025, 7, 15, 5, 29, 8, 459749, tzinfo=datetime.timezone.utc), content='Your last message was a greeting: "Hi, my i\'m Al Amin, How are you!!" There were no previous questions from you. Let\'s start fresh! What can I help you with today?', type='TextMessage')], stop_reason=None)

TaskResult(messages=[TextMessage(id='a9419b76-770d-45a2-88ce-b8f8c58cee23', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 15, 5, 29, 5, 497285, tzinfo=datetime.timezone.utc), content='What was my last question?', type='TextMessage'),

 TextMessage(id='3da3b1a9-ed9f-4a0b-b385-fb2d6b7ccf60', source='My_Assistant', models_usage=RequestUsage(prompt_tokens=163, completion_tokens=44), metadata={}, created_at=datetime.datetime(2025, 7, 15, 5, 29, 8, 459749, tzinfo=datetime.timezone.utc),
 
  content='Your last message was a greeting: "Hi, my i\'m Al Amin, How are you!!" There were no previous questions from you. Let\'s start fresh! What can I help you with today?', type='TextMessage')], 
  
  stop_reason=None)